omp/__init__.py

```diff
@@ -87,29 +99,31 @@ def init_not_msvc(self):
             except (OSError, CalledProcessError):
                 pass
 
-        # Try to load find libgomp shared library using loader search dirs
-        libgomp_path = find_library("gomp")
-
-        # Try to use custom paths if lookup failed
-        for path in paths:
-            if libgomp_path:
-                break
-            path = path.strip()
-            if os.path.isdir(path):
-                libgomp_path = find_library(os.path.join(str(path), "libgomp"))
-
-        if not libgomp_path:
-            raise ImportError("I can't find a shared library for libgomp,"
-                              " you may need to install it or adjust the "
-                              "LD_LIBRARY_PATH environment variable.")
-        else:
-            # Load the library
-            try:
-                self.libomp = ctypes.CDLL(libgomp_path)
-            except OSError:
-                raise ImportError("found openMP library '{}' but couldn't load it. "
-                                  "This may happen if you are cross-compiling.".format(libgomp_path))
-            self.version = 45
+        for libomp_name in libomp_names:
+            # Try to load find libomp shared library using loader search dirs
+            libomp_path = find_library(libomp_name)
+
+            # Try to use custom paths if lookup failed
+            for path in paths:
+                if libomp_path:
+                    break
+                path = path.strip()
+                if os.path.isfile(os.path.join(path, libomp_name)):
+                    libomp_path = os.path.join(path, libomp_name)
+
+            if libomp_path:
+                # Load the library
+                try:
+                    self.libomp = ctypes.CDLL(libomp_path)
+                except OSError:
+                    raise ImportError("found openMP library '{}' but couldn't load it. "
+                                      "This may happen if you are cross-compiling.".format(libomp_path))
+                self.version = 45
+                return
+
+        raise ImportError("I can't find a shared library for libomp, you may need to install it "
+                          "or adjust the {} environment variable.".format(env_vars[O]))
```

Comment:
The problem was trivial, this should be `0` (zero) not `O`. Fix in gh-1887.

Reply:
Thanks for the fix, looks like that slipped in after my testing.